### DISCUSSION TITLE
How to fix Error 400: redirect_uri_mismatch?

### DISCUSSION THREAD

**Discussion by JaybeRF** [YorkshireIoT/ha-google-fit#174] [Q&A] [answered]
I used this integration for some time ago, then I was a little bit disappointed with data updating so slowly, When integration was updated I try to reconnect it to Google fit with credentials which already worked. I added my HA Uri to app uri in Google console but still constantly get  Error 400: redirect_uri_mismatch when trying to connect.

**Comment by YorkshireIoT** [comment]
Hi @JaybeRF,

Could you post a screenshot of what the credentials page on your Google API cloud account looks like (with you client ID + secret hidden)?

**Reply by JaybeRF** [reply]
Hello, here it is
![123](https://github.com/YorkshireIoT/ha-google-fit/assets/29609561/ca118746-25bb-4b07-acae-c2c5f8134608)

At the time I created these credentials, everything worked well

**Reply by YorkshireIoT** [reply, marked as answer]
That looks like you've selected the wrong application type. Can you follow the instructions in the README which points you [here](https://next.home-assistant.io/integrations/google_mail/)? Specifically you need to be sure to select Web application and put in the exact redirect URI it has in the instructions

**Reply by JaybeRF** [reply]
So everything changed in settings so it was needed to create everything from a scratch and not to try to reuse
